crazy_rl/multi_agent/jax/catch/catch.py

```diff
@@ -0,0 +1,273 @@
+                    jnp.array(
+                        [jnp.delete(state.agents_locations, agent, axis=0).flatten() for agent in range(self.num_drones)]
+                    ),
+                    axis=1,
+                ),
+                target_location=target_location,
+            ),
+            key,
+        )
+
+    @override
+    @partial(jit, static_argnums=(0,))
+    def _compute_action(self, state, actions):
+        # Actions are clipped to stay in the map and scaled to do max 20cm in one step
+        return jdc.replace(
+            state,
+            agents_locations=jnp.clip(
+                state.agents_locations + actions * 0.2, jnp.array([-self.size, -self.size, 0]), self.size
+            ),
+        )
+
+    @override
+    @partial(jit, static_argnums=(0,))
+    def _compute_reward(self, state):
+        # Reward is the mean distance to the other agents plus a maximum value minus the distance to the target
+
+        return jdc.replace(
+            state,
+            rewards=jnp.any(state.truncations)
+            * (
+                # mean distance to the other agents
+                jnp.array(
+                    [
+                        jnp.sum(self.norm(state.agents_locations[agent] - state.agents_locations))
+                        for agent in range(self.num_drones)
+                    ]
+                )
+                * 0.05
+                / (self.num_drones - 1)
+                # a maximum value minus the distance to the target
+                + 0.95 * (2 * self.size - self.norm(state.agents_locations - state.target_location))
+            )
+            # negative reward if the drones crash
+            + jnp.any(state.terminations) * -10 * jnp.ones(self.num_drones),
+        )
+
+    @override
+    @partial(jit, static_argnums=(0,))
+    def _compute_terminated(self, state):
+        # collision with the ground and the target
+        terminated = jnp.logical_or(
+            state.agents_locations[:, 2] < 0.2, jnp.linalg.norm(state.agents_locations - state.target_location) < 0.2
+        )
+
+        for agent in range(self.num_drones):
+            distances = self.norm(state.agents_locations[agent] - state.agents_locations)
+
+            # collision between two drones
+            terminated = terminated.at[agent].set(
+                jnp.logical_or(terminated[agent], jnp.any(jnp.logical_and(distances > 0.001, distances < 0.2)))
+            )
+
+        return jdc.replace(
+            state, terminations=((jnp.any(state.truncations) - 1) * jnp.any(terminated)) * jnp.ones(self.num_drones)
+        )
+
+    @override
+    @partial(jit, static_argnums=(0,))
+    def _compute_truncation(self, state):
+        return jdc.replace(state, truncations=(state.timestep == 100) * jnp.ones(self.num_drones))
+
+    @override
+    @partial(jit, static_argnums=(0,))
+    def _initialize_state(self):
+        return State(
+            self._init_flying_pos,
+            0,
+            jnp.array([]),
```

Comment:
This should contain the initial positions of the drones no?

Reply:
Also, please use named arguments for such constructors, it is easier to know what is what